### PR TITLE
label, tune: Don't accept invalid serial numbers

### LINKED ISSUE #228: `tune.exfat` and `exfatlabel` don't print an error when setting an invalid volume serial
Both `tune.exfat` and `exfatlabel` don't print an error when setting an invalid volume serial.

exfatprogs version : 1.2.1

Create a container:
```
fallocate -l 1G ./mycontainer
```

Format it as exFAT:
```
mkfs.exfat ./mycontainer
```

Display volume serial:
```
tune.exfat -i ./mycontainer | grep serial
```

should give something like `volume serial : 0x66fb7876`

Attempt to set an invalid serial (serial numbers are supposed to be hex digits):
```
tune.exfat -I 0x123qq678 ./mycontainer
```
The command completes successfully. `tune.exfat -i ./mycontainer | grep serial` reports `volume serial : 0x123` which shows that the serial was truncated. The command should instead report an error and not proceed with setting any serial number.

Additionally, if you try to set a too long serial number:
```
tune.exfat -I 0x1234567890abcd ./mycontainer
```

then `tune.exfat -i ./mycontainer | grep serial` will give `volume serial : 0x7890abcd` which is the last 8 digits of the serial that was attempted to be set. The command should instead report an error and not proceed with setting any serial number.

### PR DESCRIPTION
**Pull request by lvxnull**
Fixes #228

### CODE REVIEW

File: label/label.c [hunk 1]
```diff
@@ -92,7 +93,16 @@ int main(int argc, char *argv[])
...
 			ret = exfat_show_volume_serial(bd.dev_fd);
 		} else if (flags == EXFAT_SET_VOLUME_SERIAL) {
-			ui.volume_serial = strtoul(argv[3], NULL, 0);
+			ret = exfat_parse_ulong(argv[3], &volume_serial);
+			if (volume_serial > UINT_MAX) {
+				ret = -ERANGE;
+			}
+
+			if (ret < 0) {
+				exfat_err("invalid serial number(%s)\n", argv[3]);
```
Comment (namjaejeon) [inline label/label.c:102]: we should jump to go out not to call exfat_set_volume_serial() ?

**Comment by namjaejeon** [comment]
@lvxnull Looks good:) BTW can you add prefix to subject ? and need to add the patch description.

File: lib/libexfat.c [hunk 2]
```diff
@@ -1039,3 +1039,19 @@ int read_boot_sect(struct exfat_blk_dev *bdev, struct pbr **bs)
 	free(pbr);
 	return err;
 }
+
+int exfat_parse_ulong(const char *s, unsigned long *out)
+{
+	char *endptr;
+	errno = 0;
```
Comment (namjaejeon) [inline lib/libexfat.c:1046]: this is really needed ?

Comment (lvxnull) [reply lib/libexfat.c:1046]: I'm not sure.

Comment (namjaejeon) [reply lib/libexfat.c:1046]: If so, please remove this line.

**Comment by namjaejeon** [comment]
Applied it to #exfat-next. Thanks!

**Pull request closed without merging**